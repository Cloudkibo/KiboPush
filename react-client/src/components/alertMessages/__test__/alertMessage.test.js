import React from 'react'
import ReactDOM from 'react-dom'
import AlertMessage from './../alertMessage'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AlertMessage></AlertMessage>, div)
})

it('renders alert message correctly for page type', () => {
      const {getByTestId} = render(<AlertMessage type='page'></AlertMessage>)
      expect(getByTestId('alertMessage')).toHaveTextContent("0 Pages Connected! You have no pages connected. Please connect your facebook page to use this feature.")
})

it('renders alert message correctly for subscriber type', () => {
    const {getByTestId} = render(<AlertMessage type='subscriber'></AlertMessage>)
    expect(getByTestId('alertMessage')).toHaveTextContent("0 Subscribers! Your connected pages have zero subscribers. Unless you do not have any subscriber, you will not be able to broadcast message, polls and surveys.")
})

it('matches snapshot for type page', () => {
    const tree = renderer.create(<AlertMessage type='page'></AlertMessage>).toJSON()
    expect(tree).toMatchSnapshot() 
})

it('matches snapshot for type subscriber', () => {
    const tree = renderer.create(<AlertMessage type='subscriber'></AlertMessage>).toJSON()
    expect(tree).toMatchSnapshot() 
})