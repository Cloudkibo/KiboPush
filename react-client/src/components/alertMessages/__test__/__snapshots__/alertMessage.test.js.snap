// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for type page 1`] = `
<div
  data-testid="alertMessage"
>
  <div
    className="m-alert m-alert--icon m-alert--icon-solid m-alert--outline alert alert-warning alert-dismissible fade show"
    role="alert"
  >
    <div
      className="m-alert__icon"
    >
      <i
        className="flaticon-exclamation-1"
        style={
          Object {
            "color": "white",
          }
        }
      />
      <span />
    </div>
    <div
      className="m-alert__text"
    >
      <strong>
        0 Pages Connected! 
      </strong>
      You have no pages connected. Please connect your facebook page to use this feature.  
    </div>
  </div>
</div>
`;

exports[`matches snapshot for type subscriber 1`] = `
<div
  data-testid="alertMessage"
>
  <div
    className="m-alert m-alert--icon m-alert--icon-solid m-alert--outline alert alert-warning alert-dismissible fade show"
    role="alert"
  >
    <div
      className="m-alert__icon"
    >
      <i
        className="flaticon-exclamation-1"
        style={
          Object {
            "color": "white",
          }
        }
      />
      <span />
    </div>
    <div
      className="m-alert__text"
    >
      <strong>
        0 Subscribers! 
      </strong>
      Your connected pages have zero subscribers. Unless you do not have any subscriber, you will not be able to broadcast message, polls and surveys.
    </div>
  </div>
</div>
`;
